app/pages/Onboarding/SelectWalletType/index.js

```diff
@@ -0,0 +1,188 @@
+      n: '',
+      mError: '',
+      nError: ''
+    };
+  }
+
+  onSubmit = () => {
+    const {isMultisig, m, n} = this.state;
+    const {t} = this.context;
+
+    if (!isMultisig) {
+      this.props.onNext(1, 1);
+      return;
+    }
+
+    const mNum = parseInt(m);
+    const nNum = parseInt(n);
+
+    if (isNaN(mNum) || mNum <= 0) {
+      this.setState({
+        mError: t('obSelectWalletTypeErrorPositive'),
+      });
+      return;
+    }
+    if (isNaN(mNum) || nNum <= 0) {
+      this.setState({
+        nError: t('obSelectWalletTypeErrorPositive'),
+      });
+      return;
+    }
+
+    if (mNum > nNum) {
+      this.setState({
+        mError: t('obSelectWalletTypeErrorLimits'),
+        nError: ''
+      });
+      return;
+    }
+
+    this.props.onNext(mNum, nNum);
+    return;
+  };
+
+  isValid = () => {
+    const {isMultisig, m, n} = this.state;
+
+    if (!isMultisig) return true;
+    if (!m || !n) return false;
+    return true;
+  }
+
+  onChange = name => e => {
+    this.setState({
+      [name]: e.target.value,
+      mError: '',
+      nError: ''
+    });
+  };
+
+  renderError(key) {
+    const err = this.state[key];
+    let msg = '';
+    if (err && err !== HIGHLIGHT_ONLY) {
+      msg = err;
+    }
+
+    return <div className="swt__error">{msg}</div>;
+  }
+
+  render() {
+    const {t} = this.context;
+    const {currentStep, totalSteps, onBack, onCancel} = this.props;
+    const {mError, nError, isMultisig, m, n} = this.state;
+
+    return (
+      <div className="swt">
+        <WizardHeader
+          currentStep={currentStep}
+          totalSteps={totalSteps}
+          onBack={onBack}
+          onCancel={onCancel}
+        />
+        <div className="swt__content">
+          <Submittable onSubmit={this.onSubmit}>
+            <div className="swt__header-text">
+              {t('obSelectWalletTypeTitle')}
+            </div>
+            <div className="swt__body-text">
+              {t('obSelectWalletTypeBody1')}<br />
+              {t('obSelectWalletTypeBody2')}
+            </div>
+
+            <div className="swt__input checkbox-wrapper">
+              <span className="import_checkbox_container">
+                <Checkbox
+                  checked={isMultisig}
+                  onChange={() => this.setState({isMultisig: !isMultisig})}
+                />
+              </span>
+              <span className="import_checkbox_text">
+                {t('obSelectWalletTypeCheckbox')}
+              </span>
+            </div>
+
+            {isMultisig &&
+              <>
+                <div className="swt__input-container">
+                  <div
+                    className={c('swt__input', {
+                      'swt__input--error': mError
+                    })}
+                  >
+                    <input
+                      type="number"
+                      placeholder={t('obSelectWalletTypeMinimum')}
+                      min={1}
+                      max={15}
```

Comment:
I found this max of 15 easy to bypass. I didn't even get a warning unless I hovered the mouse over the value. You can easily create a 1-of-100 multisig from this form. bob and hsd dont complain even though that redeem script (with 100 pubkeys) wouldnt fit in a witness item.

I even was able to create a wallet with n=1000000 and hsd threw silently in the background:

```
Sending IPC method error. {
  jsonrpc: '2.0',
  method: 'Wallet.createNewWallet',
  params: [ '<SANITIZED>' ],
  id: 347
} AssertionError [ERR_ASSERTION]: Assertion failed.
    at Account.fromOptions (/Users/matthewzipkin/Desktop/work/bob-wallet/node_modules/hsd/lib/wallet/account.js:116:7)
    at Function.fromOptions (/Users/matthewzipkin/Desktop/work/bob-wallet/node_modules/hsd/lib/wallet/account.js:164:26)
    at Wallet._createAccount (/Users/matthewzipkin/Desktop/work/bob-wallet/node_modules/hsd/lib/wallet/wallet.js:635:29)
    at Wallet.init (/Users/matthewzipkin/Desktop/work/bob-wallet/node_modules/hsd/lib/wallet/wallet.js:193:21)
    at WalletDB._create (/Users/matthewzipkin/Desktop/work/bob-wallet/node_modules/hsd/lib/wallet/walletdb.js:1272:5)
    at WalletDB.create (/Users/matthewzipkin/Desktop/work/bob-wallet/node_modules/hsd/lib/wallet/walletdb.js:1249:14)
    at WalletService.createNewWallet (/Users/matthewzipkin/Desktop/work/bob-wallet/app/background/wallet/service.js:369:13) {
  type: 'AssertionError',
  code: 'ERR_ASSERTION',
  generatedMessage: true,
  actual: false,
  expected: true,
  operator: '=='
}
```

Reply:
oops I shouldve made this comment on `n` not `m`. I think the only rule on `m` is that it is <= `n`... if you can fit all the pubkeys you want in a script then you can always fit the signatures (pretty sure)